syntheseus/search/graph/route.py

```diff
@@ -61,5 +61,18 @@ def expand_with_reactions(
     ) -> Sequence[BackwardReaction]:
         raise NotImplementedError
 
+    def get_starting_molecules(self) -> set[Molecule]:
+        """
+        Get the 'starting molecules' for this route,
+        i.e. reactant molecules which are not a product of a child reaction.
+        """
+        output: set[Molecule] = set()
+        for rxn in self._graph.nodes:
+            successor_products = {rxn.product for rxn in self.successors(rxn)}
```

Comment:
So `rxn` from the outer scope is passed as argument to `successors` and then shadowed by another variable called `rxn`? I guess the behaviour is correct but perhaps can be a bit confusing to read?

Reply:
Fixed!